tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/InterfaceConceptTest.xtend

```diff
@@ -1,113 +1,187 @@
+			assertNotNull(pcmPackage.eContents)
+
+			assertElementsEqual(umlPackage, pcmPackage)
+		]
 	}
 
-	def protected checkInterfaceConcept(OperationInterface pcmInterface) {
-		val umlInterface = helper.getModifiableCorr(pcmInterface, Interface, TagLiterals.INTERFACE_TO_INTERFACE)
-		checkInterfaceConcept(pcmInterface, umlInterface)
+	@Test
+	def void testCreateInterfaceConcept_PCM() {
+		initPCMRepository()
+
+		changePcmView [
+			var mPcmInterface = RepositoryFactory.eINSTANCE.createOperationInterface
+			mPcmInterface.entityName = TEST_INTERFACE_NAME
+			defaultPcmRepository.interfaces__Repository += mPcmInterface
+		]
+
+		validateUmlAndPcmPackagesView [
+			val umlPackage = defaultUmlModel.claimPackage(PACKAGE_NAME)
+			val pcmPackage = claimPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+
+			assertNotNull(umlContractsPackage.ownedElements)
+			assertNotNull(pcmPackage.eContents)
+
+			assertElementsEqual(umlPackage, pcmPackage)
+		]
 	}
 
-	def protected checkInterfaceConcept(Interface umlInterface) {
-		val pcmInterface = helper.getModifiableCorr(umlInterface, OperationInterface,
-			TagLiterals.INTERFACE_TO_INTERFACE)
-		checkInterfaceConcept(pcmInterface, umlInterface)
+	@Test
+	def void testRenameInterfaceConcept_UML() {
+		initUMLModel()
+
+		changeUmlView [
+			umlContractsPackage.createOwnedInterface(TEST_INTERFACE_NAME)
+			assertNotNull(umlContractsPackage.claimInterface(TEST_INTERFACE_NAME))
+		]
+
+		changeUmlView[
+			umlContractsPackage.claimInterface(TEST_INTERFACE_NAME).name = NEW_TEST_INTERFACE_NAME
+		]
+
+		validateUmlAndPcmPackagesView [
+			val umlPackage = defaultUmlModel.claimPackage(PACKAGE_NAME)
+			val pcmPackage = claimPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+
+			assertNotNull(umlContractsPackage.ownedElements)
+			assertNotNull(umlContractsPackage.claimInterface(NEW_TEST_INTERFACE_NAME))
+			assertNotNull(pcmPackage.eContents)
+
+			assertElementsEqual(umlPackage, pcmPackage)
+		]
 	}
 
-	def private Repository createRepositoryConcept() {
-		val pcmRepository = helper.createRepository
+	@Test
+	def void testRenameInterfaceConcept_PCM() {
+		initPCMRepository()
+
+		changePcmView [
+			var pcmInterface = RepositoryFactory.eINSTANCE.createOperationInterface
+			pcmInterface.entityName = TEST_INTERFACE_NAME
+			defaultPcmRepository.interfaces__Repository += pcmInterface
+			assertEquals(1, defaultPcmRepository.interfaces__Repository.size,
+				"There should be exactly one element in list.")
+			assertEquals(TEST_INTERFACE_NAME,
+				(defaultPcmRepository.interfaces__Repository.get(0) as OperationInterface).entityName)
+		]
 
-		userInteraction.addNextTextInput(PcmUmlClassApplicationTestHelper.UML_MODEL_FILE)
-		resourceAt(Path.of(PcmUmlClassApplicationTestHelper.PCM_MODEL_FILE)).startRecordingChanges => [
-			contents += pcmRepository
+		changePcmView [
+			(defaultPcmRepository.interfaces__Repository.
+				get(0) as OperationInterface).entityName = NEW_TEST_INTERFACE_NAME
```

Comment:
Common practice in other view-based test cases is to validate the other view after each change. Here, you change the pcm view twice in a row.
This applies to various other locations too.

Reply:
Added separate block for validating test preconditions in https://github.com/vitruv-tools/Vitruv-CaseStudies/pull/214/commits/4e8d7087f1e4299f14baf41d0f47c5fdac07dc35